storage-node/src/cache/lru_k.rs

```diff
@@ -0,0 +1,266 @@
+                k_dist = if history.len() < self.k {
+                    std::i32::MAX
+                } else {
+                    self.curr_timestamp - kth_timestamp
+                };
+                if (k_dist > max_k_dist)
+                    || (k_dist == max_k_dist && *kth_timestamp < earliest_timestamp)
+                {
+                    found = true;
+                    max_k_dist = k_dist;
+                    earliest_timestamp = *kth_timestamp;
+                    key_to_evict = Some(key.clone());
+                }
+            }
+        }
+        if found {
+            if let Some(key) = key_to_evict {
+                println!("-------- Evicting Key: {:?} --------", key);
+                if let Some(node) = self.cache_map.remove(&key) {
+                    self.size -= node.value.size();
+                }
+                return Some(key);
+            }
+        }
+        None
+    }
+
+    fn record_access(&mut self, node: &mut LruKNode<V>) {
+        node.history.push_back(self.curr_timestamp);
+        if node.history.len() > self.k {
+            node.history.pop_front();
+        }
+        self.curr_timestamp += 1;
+    }
+
+    fn get_value(&mut self, key: &K) -> Option<&V> {
+        if let Some(mut node) = self.cache_map.remove(key) {
+            self.record_access(&mut node);
+            self.cache_map.insert(key.clone(), node);
+            if let Some(cache_value) = self.cache_map.get(key).map(|node| &node.value) {
+                return Some(cache_value);
+            }
+        }
+        None
+    }
+
+    fn put_value(&mut self, key: K, value: V) -> bool {
+        if value.size() > self.max_capacity {
+            // If the object size is greater than the max capacity, we do not insert the
+            // object into the cache.
+            println!("Warning: The size of the value is greater than the max capacity",);
+            println!(
+                "Key: {:?}, Value: {:?}, Value size: {:?}, Max capacity: {:?}",
+                key,
+                value.as_value(),
+                value.size(),
+                self.max_capacity
+            );
+            return false;
+        }
+        let updated_size = value.size();
+        if let Some(mut node) = self.cache_map.remove(&key) {
+            self.record_access(&mut node);
+            self.size -= node.value.size();
+            self.cache_map.insert(key.clone(), node);
+        } else {
+            self.cache_map.insert(
+                key.clone(),
+                LruKNode {
+                    value,
+                    history: vec![self.curr_timestamp].into(),
+                },
+            );
+            self.curr_timestamp += 1;
+        }
+        self.size += updated_size;
+        while self.size > self.max_capacity {
+            let key_to_evict = self.evict(&key);
+            if key_to_evict.is_none() {
+                panic!("Fail to evict key {:?}", key);
+            }
```

Comment:
```suggestion
            debug_assert!(
                key_to_evict.is_some(),
                "key {:?} should be evicted when cache size is greater than max capacity",
                key
            );
```
Maybe we can make the message clearer.